client/lib/media/utils.js

```diff
@@ -186,7 +192,7 @@ export function getMimeType( media ) {
 	}
 
 	extension = extension.toLowerCase();
-	if ( MimeTypes.hasOwnProperty( extension ) ) {
+	if ( Object.keys( MimeTypes ).includes( extension ) ) {
```

Comment:
what's this change for? also, it looks like we're changing what might be a more efficient method with one that creates a new array of object keys then runs a slower method on that new array…?

Reply:
The linter has now started complaining about these checks; this seemed like the most correct way of performing the same check. I could instead do `extension in MimeTypes`, but that could potentially fail if `extension` happens to be a string that matches one of the object built-ins (because `in` traverses the prototype chain).

Reply:
> As for performance, it would seem like a questionable micro-optimisation to opt for a different syntax purely for performance reasons here.

It caught my attention because it was an active change more than because it's slower. As in, sometimes a change by itself doesn't flag anything, but a change in the slower direction, especially from you might rouse suspicion 😉 

> The linter has now started complaining about these checks;

For those who are less informed, and less enthusiastic, what does the linter have to complain about this?

This is quite idiomatic JavaScript and I instantly knew what it was checking. In contrast, I had to open the JS docs to see if `Object.keys()` behaves like `getOwnPropertyNames()` or `for…in` or `for…of`.

What are your thoughts here? If the linter hadn't complained would you have made this change?

Reply:
Changed back, as the linter no longer seems to care.